### PR TITLE
MLA Health Status

### LINKED ISSUE #7027: MLA health status for KKP UI
Provide MLA health status endpoints to report MLA health status per each user cluster in KKP UI (e.g. Prometheus/Promtail running in user cluster, Alertmanager configuration successfully applied, etc.).

### PR DESCRIPTION
**Pull request by hdurand0710**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7027

**Special notes for your reviewer**:
This PR adds 2 new health status in the `Cluster` `.extendedHealth`:
- `monitoring`
- `logging`
for the backend part.
It also includes the changes in the API to get those new health statuses.

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Add MLA Health status in the API /health endpoint:
- . extendedHealth. monitoring
- . extendedHealth. logging
```

### CODE REVIEW

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go [hunk 1]
```diff
@@ -876,13 +879,33 @@ func (r *reconciler) healthCheck(ctx context.Context) error {
 	}
 	oldCluster := cluster.DeepCopy()
 
-	ctrlHealth, auditHealth, err := r.getGatekeeperHealth(ctx)
-	if err != nil {
-		return err
+	if r.opaIntegration {
+		ctrlGateKeeperHealth, auditGateKeeperHealth, err := r.getGatekeeperHealth(ctx)
```
Comment (xrstf) [inline pkg/controller/user-cluster-controller-manager/resources/reconciler.go]: Please use the same casing for Gatekeeper as we use everywhere else (i.e. no capital K)

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go [hunk 2]
```diff
@@ -915,6 +938,32 @@ func (r *reconciler) getGatekeeperHealth(ctx context.Context) (
...
+func (r *reconciler) getMLAMonitoringHealth(ctx context.Context) (
+	health kubermaticv1.HealthStatus, err error) {
+
+	health, err = resources.HealthyDeployment(ctx,
+		r.Client,
+		types.NamespacedName{Namespace: resources.UserClusterMLANamespace, Name: resources.UserClusterPrometheusDeploymentName},
+		1)
+	if err != nil {
+		return kubermaticv1.HealthStatusDown,
+			fmt.Errorf("failed to get dep health %q: %v", resources.UserClusterPrometheusDeploymentName, err)
```
Comment (xrstf) [inline pkg/controller/user-cluster-controller-manager/resources/reconciler.go]: `failed to get health for Deployment %s: %w`

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go [hunk 3]
```diff
@@ -915,6 +938,32 @@ func (r *reconciler) getGatekeeperHealth(ctx context.Context) (
...
+	return health, nil
+}
+
+func (r *reconciler) getMLALoggingHealth(ctx context.Context) (kubermaticv1.HealthStatus, error) {
+	loggingHealth, err := resources.HealthyDaemonSet(ctx,
+		r.Client,
+		types.NamespacedName{Namespace: resources.UserClusterMLANamespace, Name: resources.PromtailDaemonSetName},
+		1)
+	if err != nil {
+		return kubermaticv1.HealthStatusDown, fmt.Errorf("failed to get ds health %q: %v", resources.PromtailDaemonSetName, err)
```
Comment (xrstf) [inline pkg/controller/user-cluster-controller-manager/resources/reconciler.go]: `failed to get health for DaemonSet %s: %w`:)

File: pkg/crd/kubermatic/v1/cluster.go [hunk 4]
```diff
@@ -775,6 +775,8 @@ type ExtendedClusterHealth struct {
 	UserClusterControllerManager HealthStatus `json:"userClusterControllerManager"`
 	GatekeeperController         HealthStatus `json:"gatekeeperController,omitempty"`
 	GatekeeperAudit              HealthStatus `json:"gatekeeperAudit,omitempty"`
+	UserClusterMLAMonitoring     HealthStatus `json:"userClusterMlaMonitoring,omitempty"`
```
Comment (xrstf) [inline pkg/crd/kubermatic/v1/cluster.go]: Personally I find the "UserCluster" redundant here. The entire struct is about userclusters and we do not call it UserClusterApiserver, UserClusterScheduler, UserClusterCloudProviderInfrastructure etc.

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go (hunk 1 above)
Comment (hdurand0710) [reply pkg/controller/user-cluster-controller-manager/resources/reconciler.go]: Done.

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go (hunk 2 above)
Comment (hdurand0710) [reply pkg/controller/user-cluster-controller-manager/resources/reconciler.go]: Done

File: pkg/controller/user-cluster-controller-manager/resources/reconciler.go (hunk 3 above)
Comment (hdurand0710) [reply pkg/controller/user-cluster-controller-manager/resources/reconciler.go]: Done :-)

File: pkg/crd/kubermatic/v1/cluster.go (hunk 4 above)
Comment (hdurand0710) [reply pkg/crd/kubermatic/v1/cluster.go]: You're right. I've have update the names and re-pushed.

**Review by jiachengxu — APPROVED** [review]
/approve
/lgtm

File: pkg/crd/kubermatic/v1/cluster.go [hunk 5]
```diff
@@ -775,6 +775,8 @@ type ExtendedClusterHealth struct {
 	UserClusterControllerManager HealthStatus `json:"userClusterControllerManager"`
 	GatekeeperController         HealthStatus `json:"gatekeeperController,omitempty"`
 	GatekeeperAudit              HealthStatus `json:"gatekeeperAudit,omitempty"`
+	MlaMonitoring                HealthStatus `json:"mlaMonitoring,omitempty"`
+	MlaLogging                   HealthStatus `json:"mlaLogging,omitempty"`
```
Comment (kron4eg) [inline pkg/crd/kubermatic/v1/cluster.go]: I know @xrstf already persuaded to not use UserClusterMLAMonitoring, but let's discuss about this naming a bit even more.

MLA stands for Monitoring Logging and Alerting, so we have "MonitoringLoggingAlertingMonitoring" basically. Isn't Monitoring / Logging sufficient?

Comment (hdurand0710) [reply pkg/crd/kubermatic/v1/cluster.go]: I fully agree and will integrate this change. Done.

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**